### ISSUE TITLE
csrfToken not working in 0.11.2

### ISSUE THREAD

**Issue by dottodot** [balderdashy/sails#3287]
Using 0.11.1 making a request to csrfToken returns a token as expected but in 0.11.2 nothing happens as if the route isn't there anymore. Has something changed to the way it should be used?

**Comment by sgress454** [comment]
Try doing a fresh install of Sails:
```
npm uninstall -g sails
npm cache clean
npm install sails
```

If the problem persists, please post back with more details, including what actually happens when you hit the route--do you get a 404, or a blank response, or what?

**Comment by dottodot** [comment]
OK that's not worked. I'm not getting a 404 it just loads the frontend of the app, it's as if the route just doesn't  exist. 

I'm using a blueprint prefix of

```  prefix: '/api',```

and the following route for my single page app.

```
'/*': {
     view: 'index',
     skipAssets: true,
     skipRegex: /^\/api\/.*$/
   }
```

I've also noticed that in lib/hooks/csrf/index.js there have been some changes for example `route: '/csrfToken'` has been removed so I've tried including it in my settings but that didn't work

```
module.exports.csrf = {
   grantTokenViaAjax: true,
   origin: 'http://localhost:1337',
   route: '/csrfToken'
};
```

**Comment by sgress454** [comment]
> I've also noticed that in lib/hooks/csrf/index.js there have been some changes for example route: '/csrfToken' has been removed

That option is only available in the master (v0.12.x, unpublished) version of Sails, so you wouldn't have access to it in 11.x.

The route you have set up matches _everything_ except `/api/*` routes, so of course it won't serve `/csrfToken`.  The only way I can imagine this ever worked for you is if you had `view` set to a view that doesn't exist, in which case it apparently falls through and continues trying to match routes, but it's not a behavior I would count on.  Instead, just add the csrf route to your `skipRegex`:

```
'/*': {
     view: 'index',
     skipAssets: true,
     skipRegex: /(^\/api\/.*$)|^\/csrfToken$/
   }
```
